test/e2e/destroypod_test.go

```diff
@@ -0,0 +1,134 @@
+	if err != nil {
+		t.Fatalf("Failed to create Route and Configuration: %v", err)
+	}
+	test.CleanupOnInterrupt(func() { TearDown(clients, names, logger) }, logger)
+	defer TearDown(clients, names, logger)
+
+	logger.Info("When the Revision can have traffic routed to it, the Route is marked as Ready")
+	if err := test.WaitForRouteState(clients.ServingClient, names.Route, test.IsRouteReady, "RouteIsReady"); err != nil {
+		t.Fatalf("The Route %s was not marked as Ready to serve traffic: %v", names.Route, err)
+	}
+
+	route, err := clients.ServingClient.Routes.Get(names.Route, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Error fetching Route %s: %v", names.Route, err)
+	}
+	domain := route.Status.Domain
+
+	err = test.WaitForConfigurationState(clients.ServingClient, names.Config, func(c *v1alpha1.Configuration) (bool, error) {
+		if c.Status.LatestCreatedRevisionName != names.Revision {
+			names.Revision = c.Status.LatestCreatedRevisionName
+			return true, nil
+		}
+		return false, nil
+	}, "ConfigurationUpdatedWithRevision")
+	if err != nil {
+		t.Fatalf("Error obtaining Revision's name %v", err)
+	}
+
+	_, err = pkgTest.WaitForEndpointState(
+		clients.KubeClient,
+		logger,
+		domain,
+		pkgTest.Retrying(pkgTest.MatchesBody(timeoutExpectedOutput), http.StatusNotFound),
+		"TimeoutAppServesText",
+		test.ServingFlags.ResolvableDomain)
+	if err != nil {
+		t.Fatalf("The endpoint for Route %s at domain %s didn't serve the expected text \"%s\": %v", names.Route, domain, helloWorldExpectedOutput, err)
+	}
+
+	client, err := pkgTest.NewSpoofingClient(clients.KubeClient, logger, domain, test.ServingFlags.ResolvableDomain)
+	if err != nil {
+		t.Fatalf("Error creating spoofing client: %v", err)
+	}
+
+	// The timeout app sleeps for the time passed via the timeout query parameter in milliseconds
+	timeoutRequestDurationInMillis := timeoutRequestDuration / time.Millisecond
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("http://%s/?timeout=%d", domain, timeoutRequestDurationInMillis), nil)
+	if err != nil {
+		t.Fatalf("Error creating http request: %v", err)
+	}
+
+	g, _ := errgroup.WithContext(context.TODO())
+
+	g.Go(func() error {
+		logger.Info("Sending in a long running request")
+		res, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+
+		if res.StatusCode != 200 {
+			return fmt.Errorf("Expected response to have status 200, had %d", res.StatusCode)
+		}
+		expectedBody := fmt.Sprintf("Slept for %d milliseconds", timeoutRequestDurationInMillis)
+		gotBody := string(res.Body)
+		if gotBody != expectedBody {
+			return fmt.Errorf("Unexpected body, expected: %q got: %q", expectedBody, gotBody)
+		}
+		return nil
+	})
+
+	g.Go(func() error {
+		// Give the request a bit of time to be established and reach the pod.
+		time.Sleep(timeoutRequestDuration / 2)
+
+		logger.Info("Destroying the configuration (also destroys the pods)")
+		return clients.ServingClient.Configs.Delete(names.Config, nil)
```

Comment:
We delete the resources in Teardown() already. There is also the delete method in test/clients.go for deleting resources. Do either of those work or this is doing something different?

Reply:
This is explicitly killing the configuration (which takes the pods with it) to test that the request survives that. This is to specifically test that we survive pod killings (as opposed to routes etc. being killed). If anything we could narrow this down even further to kill just the pod. I'd not use the general Teardown methods here though, as they remove just about everything.

Reply:
ok. thanks.